src/background.ts

```diff
@@ -39,21 +33,25 @@ const buildPrompt = (prompt: string, textContent: string) => {
  */
 browser.runtime.onMessage.addListener(
   async (message: { type: MessageTypesT; data: any }) => {
-    // TODO - probably need to diversify prompts based on type of request
+    // TODO - probably need to diversify prompts based on type of request, may need to
+    // set the patter to send textContent and prompt separately into each unique service
     const prompt = buildPrompt(message.data.prompt, message.data.textContent)
 
     if (message.type === 'page_qa') {
-      const result = await getOpenAIResponse(prompt)
+      const result = await getPageAssistResponse(
+        message.data as MessagePageAssistT
+      )
       browser.runtime.sendMessage({
         type: 'ai_result',
         result: result,
       })
     }
 
     if (message.type === 'page_summarize') {
-      const result = await getMlEngineAIResponse(prompt)
-      // const result = await getLocalModelResponse(prompt)
-      // const result = await getHuggingFaceResponse(prompt)
+      const result = await getPageAssistResponse(
+        message.data as MessagePageAssistT
```

Comment:
is it intentional that both page_qa and page_summarize do the same thing now except response type (where i wonder if this should be more generic and pass on a result with type `message.type + '_result'`?

i suppose maybe related: why do we go through messages instead of having something like MozQuestionAnswer directly call getPageAssistResponse where we would still need a message to get the page content

Reply:
I do have a follow up [branch](https://github.com/Firefox-AI/extension-hub/pull/16/files#diff-11db236acd6b02b1229e7d325e9bf8edb521032e6388f68b6a884ec24bbe291f) that updates the message type, I agree the ai_result is pretty worthless naming. 

I see what you are saying maybe something more like in MozChat where I'm getting the page content in the lit component via events and then from there I don't even need background.ts as a middle man there? I guess I figured it was a nice pattern in case any of the services needed to use the privileges of background.ts and it was just ready to go..?